be/src/runtime/task_group/task_group.cpp

```diff
@@ -132,17 +148,23 @@ void TaskGroup::remove_mem_tracker_limiter(std::shared_ptr<MemTrackerLimiter> me
 }
 
 int64_t TaskGroup::memory_limit_gc() {
-    std::string name;
-    int64_t memory_limit;
-    {
-        std::shared_lock<std::shared_mutex> rl {_mutex};
-        name = _name;
-        memory_limit = _memory_limit;
-    }
-    return MemTrackerLimiter::tg_memory_limit_gc(_id, name, memory_limit,
+    const auto& [memory_limit, name] = _memory_limit_and_name();
+    auto used = memory_used();
+    return MemTrackerLimiter::tg_memory_limit_gc(used - memory_limit, used, _id, name, memory_limit,
                                                  _mem_tracker_limiter_pool);
 }
 
+int64_t TaskGroup::soft_memory_limit_gc(int64_t request_free_memory, int64_t used_memory) {
```

Comment:
Add comment, the difference between `soft_memory_limit_gc` and `memory_limit_gc` is that `soft_memory_limit_gc` request_free_memory <= used - memory_limit

maybe can change name